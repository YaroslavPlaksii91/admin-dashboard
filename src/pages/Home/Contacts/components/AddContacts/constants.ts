import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const ADD_CONTACT_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const ADD_CONTACT_CONFIG = Object.freeze({
  [ADD_CONTACT_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [ADD_CONTACT_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [ADD_CONTACT_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [ADD_CONTACT_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [ADD_CONTACT_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
