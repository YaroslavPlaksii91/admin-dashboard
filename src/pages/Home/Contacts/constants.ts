export const CONTACTS_COLUMNS = Object.freeze([
  {
    title: 'Name',
    name: 'name',
    size: 4,
  },
  {
    title: 'Email',
    name: 'email',
    size: 3,
  },
  {
    title: 'Address',
    name: 'address',
    size: 3,
  },
  {
    title: 'Created at',
    name: 'created',
    size: 2,
  },
]);
