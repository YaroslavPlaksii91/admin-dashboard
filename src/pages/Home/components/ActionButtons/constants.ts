export const SORT_BTN_TEST_ID = 'sortButton';
export const FILTER_BTN_TEST_ID = 'filterButton';
export const ADD_BTN_TEST_ID = 'addButton';
