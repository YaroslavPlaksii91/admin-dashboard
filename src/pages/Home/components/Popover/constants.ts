export const POPOVER_TEST_ID = 'popover';
