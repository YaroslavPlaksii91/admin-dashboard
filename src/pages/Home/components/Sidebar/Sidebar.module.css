.link {
  display: flex;
  align-items: center;
  width: 100%;
  padding: 18px 0 18px 32px;
  letter-spacing: 0.2px;
  transition:
    color 250ms ease,
    background-color 250ms ease;
}

.link:hover,
.link:focus {
  color: var(--accent-sidebar-color);
  background-color: rgba(159, 162, 180, 0.08);
}

.link svg {
  margin-right: 24px;
  color: var(--secondary-text-color);
  opacity: 0.4;
  transition:
    color 250ms ease,
    opacity 250ms ease;
}

.link:hover svg,
.link:focus svg,
.activeLink svg {
  color: var(--accent-sidebar-color);
  opacity: 1;
}

.activeLink {
  composes: link;
  color: var(--accent-sidebar-color);
  background-color: rgba(159, 162, 180, 0.08);
}
