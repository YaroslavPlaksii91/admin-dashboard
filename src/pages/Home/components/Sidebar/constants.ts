export const USERS_PAGES = Object.freeze([
  {
    name: 'Overview',
    icon: '/src/assets/icons/sprite.svg#icon-overview',
    link: '/overview',
  },
  {
    name: 'Tickets',
    icon: '/src/assets/icons/sprite.svg#icon-tickets',
    link: '/tickets',
  },
  {
    name: 'Ideas',
    icon: '/src/assets/icons/sprite.svg#icon-ideas',
    link: '/ideas',
  },
  {
    name: 'Contacts',
    icon: '/src/assets/icons/sprite.svg#icon-contacts',
    link: '/contacts',
  },
  {
    name: 'Agents',
    icon: '/src/assets/icons/sprite.svg#icon-agents',
    link: '/agents',
  },
  {
    name: 'Articles',
    icon: '/src/assets/icons/sprite.svg#icon-articles',
    link: '/articles',
  },
]);

export const USERS_SETTINGS = Object.freeze([
  {
    name: 'Settings',
    icon: '/src/assets/icons/sprite.svg#icon-settings',
    link: '/settings',
  },
  {
    name: 'Subscription',
    icon: '/src/assets/icons/sprite.svg#icon-subscription',
    link: '/subscription',
  },
]);

export const SIDEBAR_LINK_TEST_ID = 'sidebarLink';
