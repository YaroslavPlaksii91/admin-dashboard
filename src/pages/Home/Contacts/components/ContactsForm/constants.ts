import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const CONTACTS_FORM_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const CONTACTS_FORM_CONFIG = Object.freeze({
  [CONTACTS_FORM_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [CONTACTS_FORM_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [CONTACTS_FORM_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [CONTACTS_FORM_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [CONTACTS_FORM_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
