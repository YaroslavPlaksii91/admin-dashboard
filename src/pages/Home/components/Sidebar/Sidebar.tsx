import { NavLink } from 'react-router-dom';
import {
  SvgIcon,
  Drawer,
  List,
  ListItem,
  Typography,
  Divider,
} from '@mui/material';

import { Logo } from '@components/Logo/Logo';
import { LOGO_TYPES } from '@components/Logo/constants';

import { USERS_PAGES, USERS_SETTINGS } from './constants';

import styles from './Sidebar.module.css';

export const Sidebar = () => {
  return (
    <Drawer variant="permanent" anchor="left">
      <Logo type={LOGO_TYPES.HORIZONTAL} />

      <List sx={{ padding: 0, paddingBottom: 4 }}>
        {USERS_PAGES.map(page => (
          <ListItem key={page.name} disablePadding sx={{ letterSpacing: 0.2 }}>
            <NavLink
              to={page.link}
              className={({ isActive }) =>
                isActive ? styles.activeLink : styles.link
              }
            >
              <SvgIcon titleAccess={page.name} fontSize="small">
                <use href={page.icon}></use>
              </SvgIcon>
              <Typography
                variant="body1"
                component="span"
                sx={{ lineHeight: 1.3 }}
              >
                {page.name}
              </Typography>
            </NavLink>
          </ListItem>
        ))}
      </List>

      <Divider
        sx={{
          backgroundColor: 'borderColor',
          opacity: 0.06,
        }}
      />

      <List sx={{ padding: 0, paddingTop: 4 }}>
        {USERS_SETTINGS.map(page => (
          <ListItem key={page.name} disablePadding sx={{ letterSpacing: 0.2 }}>
            <NavLink
              to={page.link}
              className={({ isActive }) =>
                isActive ? styles.activeLink : styles.link
              }
            >
              <SvgIcon titleAccess={page.name} fontSize="small">
                <use href={page.icon}></use>
              </SvgIcon>
              <Typography
                variant="body1"
                component="span"
                sx={{ lineHeight: 1.3 }}
              >
                {page.name}
              </Typography>
            </NavLink>
          </ListItem>
        ))}
      </List>
    </Drawer>
  );
};
