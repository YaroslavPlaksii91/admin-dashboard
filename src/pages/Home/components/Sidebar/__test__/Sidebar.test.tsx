import { cleanup, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { Sidebar } from '../Sidebar';
import { SIDEBAR_LINK_TEST_ID } from '../constants';

describe('Sidebar component', () => {
  afterEach(cleanup);

  it('should be rendered', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Sidebar />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(
      <BrowserRouter>
        <Sidebar />
      </BrowserRouter>,
    );

    const logo = screen.getByText('Dashboard Kit');
    expect(logo).toBeInTheDocument();

    const links = screen.getAllByTestId(SIDEBAR_LINK_TEST_ID);

    links.forEach(link => {
      expect(link).toBeInTheDocument();
    });
  });
});
