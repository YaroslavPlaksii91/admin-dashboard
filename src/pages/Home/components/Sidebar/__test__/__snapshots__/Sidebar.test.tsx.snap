// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        class="undefined"
      >
        <a
          href="/"
        >
          <svg
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium undefined css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
          >
            <use
              href="/sprite.svg#icon-logo"
            />
            <title>
              Logo
            </title>
          </svg>
          <p
            class="MuiTypography-root MuiTypography-h3 css-14qosaq-MuiTypography-root"
          >
            Dashboard Kit
          </p>
        </a>
      </div>
      <ul
        class="MuiList-root MuiList-padding css-ceqh33-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/overview"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-overview"
              />
              <title>
                Overview
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Overview
            </span>
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/tickets"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-tickets"
              />
              <title>
                Tickets
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Tickets
            </span>
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/ideas"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-ideas"
              />
              <title>
                Ideas
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Ideas
            </span>
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/contacts"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-contacts"
              />
              <title>
                Contacts
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Contacts
            </span>
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/agents"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-agents"
              />
              <title>
                Agents
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Agents
            </span>
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/articles"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-articles"
              />
              <title>
                Articles
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Articles
            </span>
          </a>
        </li>
      </ul>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-gpu97-MuiDivider-root"
      />
      <ul
        class="MuiList-root MuiList-padding css-1k9zc9v-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/settings"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-settings"
              />
              <title>
                Settings
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Settings
            </span>
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-zjos0a-MuiListItem-root"
        >
          <a
            data-testid="sidebarLink"
            href="/subscription"
          >
            <svg
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              focusable="false"
              role="img"
              viewBox="0 0 24 24"
            >
              <use
                href="/sprite.svg#icon-subscription"
              />
              <title>
                Subscription
              </title>
            </svg>
            <span
              class="MuiTypography-root MuiTypography-body1 css-lfdebc-MuiTypography-root"
            >
              Subscription
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
